app/forms/idv/finance_form.rb

```diff
@@ -4,49 +4,70 @@ class FinanceForm
 
     FINANCE_TYPES = [:ccn, :mortgage, :home_equity_line, :auto_loan].freeze
 
-    attr_reader :idv_params, :finance_type, :finance_account
+    FINANCE_HTML_OPTIONS = {
+      ccn: { maxlength: 8 }
+    }
 
-    validates :finance_type, :finance_account, presence: true
+    attr_reader :idv_params, :finance_type, *FINANCE_TYPES
 
-    validates :finance_account,
+    validates :finance_type, presence: true
+
+    FINANCE_TYPES.each do |type|
+      validates type, presence: true, if: ->(form) { form.finance_type == type }
+    end
+
+    validates :ccn,
               format: { with: /\A\d{8}\z/, message: I18n.t('idv.errors.invalid_ccn') },
               if: :ccn?
 
-    validate :finance_type_valid
+    validate :validate_finance_type
```

Comment:
I wanted to add `valid_finance_type?` as aboolean returning, non-side-effecty method. But I was worried that having both `finance_type_valid` and `valid_finance_type?` might be confusing.

So then I figured I'd rename the validation metho , and having an action in the name (validate) seemed like the clearest way to me that the method was going to have side effects and do the validation?

But happy to move the names